DQM/HcalTasks/plugins/ZDCQIE10Task.cc

```diff
@@ -1,116 +1,579 @@
+        _cADC_vs_TS_PM[0]->Fill(i, digi[i].adc());
+        sample[0][i] = constants::adc2fC[digi[i].adc()];
+        if (i == 1) {
+          sample_ZDCm_TS1 = constants::adc2fC[digi[i].adc()];
+        }
+        if (i == 2) {
+          sample_ZDCm_TS2 = constants::adc2fC[digi[i].adc()];
+        }
+      }
+
       // iter over all samples
       if (_cADC_EChannel.find(did()) != _cADC_EChannel.end()) {
         _cADC_EChannel[did()]->Fill(digi[i].adc());
+        _cfC_EChannel[did()]->Fill(constants::adc2fC[digi[i].adc()]);
+        _cTDC_EChannel[did()]->Fill(digi[i].le_tdc());
       }
       if (_cADC_vs_TS_EChannel.find(did()) != _cADC_vs_TS_EChannel.end()) {
         _cADC_vs_TS_EChannel[did()]->Fill(i, digi[i].adc());
+        _cfC_vs_TS_EChannel[did()]->Fill(i, constants::adc2fC[digi[i].adc()]);
+      }
+      _cZDC_CapIDS[0]->Fill(digi[i].capid());
+    }
+    if (did.section() == 1) {
+      if ((did.channel() > 0) && (did.channel() < 6)) {
+        EMM_sum += (sample_ZDCm_TS2 - sample_ZDCm_TS1);
+        EMP_sum += (sample_ZDCp_TS2 - sample_ZDCp_TS1);
+        for (int k = 0; k < 6; k++) {
+          EMP_tot_sum += sample[1][k];
+          EMM_tot_sum += sample[0][k];
+        }
+        if (did.zside() == -1) {
+          if (sample[0][2] > (0.0 * EMM_tot_sum)) {
+            _cZDC_EM_TM[0]->Fill((did.channel() * 2) - 1);
+          } else {
+            _cZDC_EM_TM[0]->Fill((did.channel() * 2));
+          }
+        }
+        if (did.zside() == 1) {
+          if (sample[1][2] > (0.0 * EMP_tot_sum)) {
+            _cZDC_EM_TM[1]->Fill((did.channel() * 2) - 1);
+          } else {
+            _cZDC_EM_TM[1]->Fill((did.channel() * 2));
+          }
+        }
+      }
+    }
+    if (did.section() == 2) {
+      if ((did.channel() > 0) && (did.channel() < 5)) {
+        HADP_sum += (sample_ZDCp_TS2 - sample_ZDCp_TS1);
+        HADM_sum += (sample_ZDCm_TS2 - sample_ZDCm_TS1);
+        for (int k = 0; k < 6; k++) {
+          HADP_tot_sum += sample[1][k];
+          HADM_tot_sum += sample[0][k];
+        }
+        if (did.zside() == -1) {
+          if (sample[0][2] > (0.0 * HADM_tot_sum)) {
+            _cZDC_HAD_TM[0]->Fill((did.channel() * 2) - 1);
+          } else {
+            _cZDC_HAD_TM[0]->Fill((did.channel() * 2));
+          }
+        }
+        if (did.zside() == 1) {
+          if (sample[1][2] > (0.0 * HADP_tot_sum)) {
+            _cZDC_HAD_TM[1]->Fill((did.channel() * 2) - 1);
+          } else {
+            _cZDC_HAD_TM[1]->Fill((did.channel() * 2));
+          }
+        }
       }
     }
   }
+  // Hardcoded callibrations currently
+  _cZDC_SUMS[0]->Fill(((EMM_sum * 0.1) + HADM_sum) * 0.5031);
+  _cZDC_SUMS[1]->Fill(((EMP_sum * 0.12) + HADP_sum) * 0.9397);
+  _cZDC_BXSUMS[0]->Fill(bx, ((EMM_sum * 0.1) + HADM_sum) * 0.5031);
+  _cZDC_BXSUMS[1]->Fill(bx, ((EMP_sum * 0.12) + HADP_sum) * 0.9397);
 }
+//std::shared_ptr<hcaldqm::Cache> ZDCQIE10Task::globalBeginLuminosityBlock(edm::LuminosityBlock const& lb,
```

Comment:
Are these comments here for a reason? otherwise we can probably remove.

Reply:
Going through each comment for completeness - this was addressed in https://github.com/cms-sw/cmssw/commit/eff9cc2e664dc4e8381d0140a0c4dc94965ee6d5